2025-09-12 17:24:38,165 - ERROR - 'numpy.ndarray' object has no attribute 'unique'
Traceback (most recent call last):
  File "/homes/55/bella/miniconda3/envs/msc-project-env/lib/python3.12/site-packages/pygad/pygad.py", line 1559, in cal_pop_fitness
    fitness = self.fitness_func(self, sol, sol_idx)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/scratch/local/ssd/bella/msc-project/experiments/experiment_genetic_algorithm.py", line 33, in fitness_func
    unique_elems_score = evaluate_unique_params(solution)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/scratch/local/ssd/bella/msc-project/src/msc_project/algorithms/genetic_algorithm/objectives.py", line 46, in evaluate_unique_params
    num_unique = solution.unique()
                 ^^^^^^^^^^^^^^^
AttributeError: 'numpy.ndarray' object has no attribute 'unique'
2025-09-12 17:24:38,172 - ERROR - 'numpy.ndarray' object has no attribute 'unique'
Traceback (most recent call last):
  File "/homes/55/bella/miniconda3/envs/msc-project-env/lib/python3.12/site-packages/pygad/pygad.py", line 1765, in run
    self.last_generation_fitness = self.cal_pop_fitness()
                                   ^^^^^^^^^^^^^^^^^^^^^^
  File "/homes/55/bella/miniconda3/envs/msc-project-env/lib/python3.12/site-packages/pygad/pygad.py", line 1722, in cal_pop_fitness
    raise ex
  File "/homes/55/bella/miniconda3/envs/msc-project-env/lib/python3.12/site-packages/pygad/pygad.py", line 1559, in cal_pop_fitness
    fitness = self.fitness_func(self, sol, sol_idx)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/scratch/local/ssd/bella/msc-project/experiments/experiment_genetic_algorithm.py", line 33, in fitness_func
    unique_elems_score = evaluate_unique_params(solution)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/scratch/local/ssd/bella/msc-project/src/msc_project/algorithms/genetic_algorithm/objectives.py", line 46, in evaluate_unique_params
    num_unique = solution.unique()
                 ^^^^^^^^^^^^^^^
AttributeError: 'numpy.ndarray' object has no attribute 'unique'
